src/axom/sidre/core/View.cpp

```diff
@@ -630,17 +647,43 @@ int View::getShape(int ndims, IndexType* shape) const
   }
 
   // Fill the rest of the array with zeros (when ndims > shapeSize)
-  if(ndims > shapeSize)
+  for(int i = shapeSize ; i < ndims ; ++i)
   {
-    for(int i = shapeSize ; i < ndims ; ++i)
-    {
-      shape[i] = 0;
-    }
+    shape[i] = 0;
   }
 
   return m_shape.size();
 }
 
+/*
+ *************************************************************************
+ *
+ * Return number of dimensions and fill in permutation information.
+ *
+ *************************************************************************
+ */
+int View::getPermutation(int ndims, IndexType* permutation) const
+{
+  if (static_cast<unsigned>(ndims) < m_permutation.size())
```

Comment:
Is this the correct behavior? Or should it return an array filled with ``[0,1,2,3,4...., ndims-1]`` 

(or maybe up to ``min(ndims, getNumDimensions()``?)

Reply:
This just mimics the semantics of `View::getShape`.